### DISCUSSION TITLE
Running MJPC without GUI

### DISCUSSION THREAD

**Discussion by kgimbergh** [google-deepmind/mujoco#2238] [Asking for Help]
### Intro

Hi!

I am a student at the TU Delft, I use MuJoCo for my research on balanced standing for an exoskeleton.

### My setup

I use MJPC in C++ with MuJoCo version 3.1.4 on ubuntu 22.04.

### My question

To tune the weights in my cost function, I would like to be able to run MJPC without the GUI of MuJoCo. Now I have read that headless rendering is possible, but I have not been able to accomplish it. 

The first option I saw was to install OSMesa or EGL and to set MUJOCO_GL to 'osmesa' or 'egl', respectively. Unfortunately nothing changed in doing this and the MuJoCo GUI launched as it would normally.

When trying to disable the launch of the GUI manually in say simulate.cc, I run into segmentation faults.

I have also seen that it might be possible using python, but I would prefer to keep to C++.

Is there anything that I have looked over? Is there a simple way to get headless rendering? Thanks! 

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
Please ask this in the mujoco_mpc repo
